erpnext/public/js/controllers/transaction.js

```diff
@@ -261,11 +261,19 @@ erpnext.TransactionController = erpnext.taxes_and_totals.extend({
 		if(!in_list(["Delivery Note", "Sales Invoice", "Purchase Receipt", "Purchase Invoice"], this.frm.doc.doctype)) {
 			return;
 		}
-		var me = this;
-		var inspection_type = in_list(["Purchase Receipt", "Purchase Invoice"], this.frm.doc.doctype)
+
+		const me = this;
+		if (!this.frm.is_new() && this.frm.doc.docstatus === 0) {
+			this.frm.add_custom_button(__("Quality Inspection(s)"), () => {
+				me.make_quality_inspection();
+			}, __("Create"));
+			this.frm.page.set_inner_btn_group_as_primary(__('Create'));
+		}
```

Comment:
Add this to Stock Entry too, if `inspection_required` is True. It has it's own `setup_quality_inspection`